### PR TITLE
Spatial field

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: ckanext/ozwillo_theme/fanstatic/theme_location.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+"use strict";
+
+/* theme_location
+ *
+ * This JavaScript module gets a geojson from osm given a location name
+ * TODO: Find how to manage properly the required field,
+ *       Manage the case when the user click nowhere else than validate after typing the location
+ *       For now, the request always fails, find why
```
Comment (bobeal) [inline ckanext/ozwillo_theme/fanstatic/theme_location.js]: ?

File: ckanext/ozwillo_theme/fanstatic/theme_location.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      myform.onsubmit = function(){return validateForm()};
+      var element = document.getElementById('custom_fields');
+      element.firstElementChild.firstElementChild.style.display = 'none';
+
+      $( function() {
+        $( "#field-spatial-name" ).autocomplete({
+          minLength: 2,
+          source: function (request, response) {
+            var query = request.term;
+            var url = 'https://search.osmnames.org/fr/q/' + query + '.js?key=xS5b3xnPgEUW5zi0GvWa';
```
Comment (bobeal) [inline ckanext/ozwillo_theme/fanstatic/theme_location.js]: what is this key ? could it be externalized ?

File: ckanext/ozwillo_theme/fanstatic/theme_location.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        });
+      });
+
+      function nominatim(city) {
+        var url = 'https://nominatim.openstreetmap.org/search/' + city + '?polygon_geojson=1';
+        $.getJSON(url, {
+          format:"json"
+        })
+        .done(function(data) {
+          console.log(data);
```
Comment (bobeal) [inline ckanext/ozwillo_theme/fanstatic/theme_location.js]: remove the console.log

File: ckanext/ozwillo_theme/fanstatic/theme_location.js (hunk 2 above)
Comment (ghost) [reply ckanext/ozwillo_theme/fanstatic/theme_location.js]: It is the API key for osmnames, I will add it to the config file as the parameter

File: ckanext/ozwillo_theme/fanstatic/theme_location.js (hunk 1 above)
Comment (ghost) [reply ckanext/ozwillo_theme/fanstatic/theme_location.js]: Done in the last version, except for the "required" field.
For now a pop up appears and ask the user to fill the location field, but this is done nicely in ckan with fields highligths. I couldn't get it to work here though.
If you don't want any popup I can remove it and let users be able to create datasets with no location.

File: ckanext/ozwillo_theme/plugin.py [hunk 4]
```diff
@@ -63,14 +63,17 @@ def get_helpers(self):
...
+            'ozwillo_theme_get_member_text': lambda: config.get('ckanext.ozwillo_theme.text_number_member', 'adherents SICTIAM'),
             'ozwillo_theme_get_popular_datasets': lambda: logic.get_action('package_search')({}, {"rows": 3, 'sort': 'views_total desc'})['results'],
             'ozwillo_theme_get_popular_organizations': ozwillo_theme_popular_organizations,
             'ozwillo_theme_display_date': ozwillo_theme_display_date,
             'ozwillo_theme_get_map': ozwillo_theme_get_map,
-            'ozwillo_theme_get_title': lambda: config.get('ckan.ozwillo_theme.map_title', 'Example map')
+            'ozwillo_theme_get_title': lambda: config.get('ckanext.ozwillo_theme.map_title', 'Example map'),
+            'ozwillo_theme_spatial_installed': lambda: config.get('ckanext.ozwillo_theme.spatial_installed', 'False'),
+            'ozwillo_theme_osmnames_key': lambda: config.get('ckanext.ozwillo_theme.osmnames_key', 'xS5b3xnPgEUW5zi0GvWa')
+
```
Comment (bobeal) [inline ckanext/ozwillo_theme/plugin.py]: Same here for the key

File: ckanext/ozwillo_theme/fanstatic/theme_location.js (hunk 1 above)
Comment (bobeal) [reply ckanext/ozwillo_theme/fanstatic/theme_location.js]: Yes, I prefer not having a popup (and yes, a dataset may not have a location or, at least, we have to allow it).

**Pull request merged**